nevergrad/functions/olympussurfaces/core.py

```diff
@@ -0,0 +1,82 @@
+from ..base import ExperimentFunction
+from olympus import surfaces  # type: ignore
+from olympus import noises  # type: ignore
+
+
+class OlympusSurface(ExperimentFunction):
+    def __init__(
+        self, kind: str, dimension: int = 10, noise_kind: str = "GaussianNoise", noise_scale: float = 1
+    ) -> None:
+        self.kind = kind
+        self.param_dim = dimension
+        self.noise_kind = noise_kind
+        self.noise_scale = noise_scale
+        self.surface = partial(self._simulate_surface, noise=True)
+        self.surface_without_noise = partial(self._simulate_surface, noise=False)
+        parametrization = p.Array(shape=(dimension,))
+        parametrization.function.deterministic = False
+        super().__init__(self.surface, parametrization)
+
+    def _simulate_surface(self, x: np.ndarray, noise: bool = True) -> float:
+        assert self.kind in [
+            "AckleyPath",
+            "Dejong",
+            "HyperEllipsoid",
+            "Levy",
+            "Michalewicz",
+            "Rastrigin",
+            "Rosenbrock",
+            "Schwefel",
+            "StyblinskiTang",
+            "Zakharov",
+            "DiscreteAckley",
+            "DiscreteDoubleWell",
+            "DiscreteMichalewicz",
+            "LinearFunnel",
+            "NarrowFunnel",
+            "GaussianMixture",
+        ]
+
+        assert self.noise_kind in ["GaussianNoise", "UniformNoise", "GammaNoise"]
+
+        traditional_surfaces = {
+            "Michalewicz": surfaces.Michalewicz,
+            "AckleyPath": surfaces.AckleyPath,
+            "Dejong": surfaces.Dejong,
+            "HyperEllipsoid": surfaces.HyperEllipsoid,
+            "Levy": surfaces.Levy,
+            "Michalewicz": surfaces.Michalewicz,
+            "Rastrigin": surfaces.Rastrigin,
+            "Rosenbrock": surfaces.Rosenbrock,
+            "Schwefel": surfaces.Schwefel,
+            "StyblinskiTang": surfaces.StyblinskiTang,
+            "Zakharov": surfaces.Zakharov,
+            "DiscreteAckley": surfaces.DiscreteAckley,
+            "DiscreteDoubleWell": surfaces.DiscreteDoubleWell,
+            "DiscreteMichalewicz": surfaces.DiscreteMichalewicz,
+            "LinearFunnel": surfaces.LinearFunnel,
+            "NarrowFunnel": surfaces.NarrowFunnel,
+            "GaussianMixture": surfaces.GaussianMixture,
+        }
+        if noise:
+            noise = noises.Noise(kind=self.noise_kind, scale=self.noise_scale)
+            surface = traditional_surfaces[self.kind](param_dim=self.param_dim, noise=noise)
+        else:
+            surface = traditional_surfaces[self.kind](param_dim=self.param_dim)
+        return surface.run(x)[0][0]
+
+    def evaluation_function(self, *recommendations) -> float:
+        """Averages multiple evaluations if necessary."""
+        x = recommendations[0].value
+        losses = [self.surface_without_noise(x) for _ in range(42)]
+        return sum(losses) / len(losses)
```

Comment:
would that work to use the non-noisy case instead?
```python
traditional_surfaces[self.kind](param_dim=self.param_dim)
return surface.run(x)[0][0]
```

Reply:
Using the non-noisy surface in the evaluation function was proposed by Olivier.

Reply:
Yes! But now we can remove the loop.

Reply:
ok got it